src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx

```diff
@@ -1998,14 +2012,16 @@ function getRequestParams(
   const valueSpec = valueSpecLookup[valueSpecConfig];
 
   // define viewport based on independent axis range: need to check undefined case
-  const viewport =
-    vizConfig?.independentAxisRange?.min != null &&
-    vizConfig?.independentAxisRange?.max != null
-      ? {
-          xMin: String(vizConfig?.independentAxisRange?.min),
-          xMax: String(vizConfig?.independentAxisRange?.max),
-        }
-      : undefined;
+  // also no viewport change regardless of the change of overlayVariable
+  const viewport = keepIndependentAxisRange
```

Comment:
I did some testing, and it seems that now there is never any `viewport` sent to the back end.  Compare this with the QA site and you'll see the 

I reverted this block of code and I get the desired behaviour, though there is an issue with missing yellow truncation bars, but that is a separate issue that I have just commented on in an old ticket: https://github.com/VEuPathDB/web-eda/issues/1295

I think we would be OK without the `keepIndependentAxisRange` state management?

Reply:
@bobular Thank you for your review and tests. Yes I intended to set not to change viewport conditionally due to two reasons: a) as you mentioned, it is due to truncation; b) another problem is that if we always use viewport but keep x-axis range, change custom range with overlay variable, and remove overlay variable, then the data itself is limited to the x-axis custom range. So, it is just not yellow truncation.

For an example of b), GEMS1 Case, Age (X), Weight (Y), Case or control participant (overlay). If one set x-axis custom range of 0 to 30, then it looks like this - looks fine for now:
![case01](https://user-images.githubusercontent.com/12802305/203099154-6afee39b-f2cd-4c9b-aa76-dc7f9bb99332.png)

However, if I remove the overlay variable, then the data is limited to 0 to 30 because we asked the data from 0 to 30 via viewport. Also, "Reset to default" would not work because data was already changed to 0 to 30 range. This was one of the reasons why I didn't want to change viewport for the change of overlay variable.
![case02](https://user-images.githubusercontent.com/12802305/203099334-0d808bb2-c091-43de-b528-9bbb61bb6394.png)